pkg/azuredisk/nodeserver.go

```diff
@@ -380,19 +380,43 @@ func (d *Driver) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandVolume
 	volSizeBytes := int64(capacityBytes)
 	requestGiB := volumehelper.RoundUpGiB(volSizeBytes)
 
-	args := []string{"-o", "source", "--noheadings", "--mountpoint", req.GetVolumePath()}
-	output, err := d.mounter.Exec.Command("findmnt", args...).Output()
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not determine device path: %v", err)
+	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume path must be provided")
 	}
 
-	devicePath := strings.TrimSpace(string(output))
-	if len(devicePath) == 0 {
-		return nil, status.Errorf(codes.Internal, "Could not get valid device for mount path: %q", req.GetVolumePath())
+	// Hopefully devicePath was passed from req
+	devicePath := req.GetStagingTargetPath()
+	if devicePath == "" {
```

Comment:
Here it should use some other logic to get the device path for volumes that are raw block for v1.19 before.

Reply:
Update: I am using isBlock to check if this is Block. After testing it seems to work